source/h2_stream.c

```diff
@@ -583,6 +607,38 @@ int aws_h2_stream_on_activated(struct aws_h2_stream *stream, bool *out_has_outgo
     return AWS_OP_ERR;
 }
 
+static void s_h2_stream_manual_write_complete(struct aws_h2_stream *stream, bool *body_stalled) {
+    AWS_PRECONDITION(body_stalled);
+
+    /* finish/clean up the current write operation */
+    struct aws_h2_stream_data_write *write_op = stream->thread_data.outgoing_write;
+    s_stream_data_write_destroy(stream, write_op);
+    stream->thread_data.outgoing_write = NULL;
+
+    s_lock_synced_data(stream);
+    {
+        /* check to see if there are more queued writes */
+        if (!aws_linked_list_empty(&stream->synced_data.pending_write_list)) {
+            /* Advance to the next write op for next time */
```

Comment:
this all looks correct.
but maybe??? simpler to just have 2 pending_write_lists and you don't need the lock here